src/router/Router.ts

```diff
@@ -47,18 +48,24 @@ export class Router {
     this.mapUrl = mapUrl;
   }
 
-  private getRecentUrl(): string {
+  private getRecentUrl(): string | null {
     if (this.recentUrl) return this.recentUrl;
     if (this.parentRouter) return this.parentRouter.getRecentUrl();
     return null;
   }
 
+  getChildUrl = () => {
+    return this.childUrl;
+  }
+
```

Comment:
Do we need to expose this ? Can't the childUrl be an internal only property ?

Reply:
@karkipy

Reply:
we do need the unprocessed url, for further processing in child router